### PR TITLE
Integrate new Stories API with SJ Client

### PR DESCRIPTION
**Pull request by UberMouse**
Acceptance Criteria
=================
- Stories endpoint can be consumed via SJ Client

Background
=================
Integrating the freshly created Stories endpoint on Supplejack into the client

Checklist
=================

- [ ] Code is understandable without Dev
- [ ] Acceptance criteria, what was changed, and why it was changed (If applicable) on Pull / Merge Request
- [ ] All Tests are Passing
- [ ] Code Coverage goes up
- [ ] All new methods have a relevant spec
- [ ] Methods have a single responsibility (Where applicable)
- [ ] Linters Pass
- [ ] ‘Yard’ style comments on methods and classes (Where applicable)

---
This change is [Reviewable](https://reviewable.io/reviews/digitalnz/supplejack_client/10)

### CODE REVIEW

File: lib/supplejack/request.rb [hunk 1]
```diff
@@ -68,6 +68,18 @@ def put(path, params={}, payload={}, options={})
       end
     end
 
+    def patch(path, params: {}, payload: {}, options: {})
+      payload ||= {}
+      log_request(:patch, path, params, payload) do
+        response = RestClient::Request.execute(:url => full_url(path, nil, params),
```
Comment (EdwinRozario) [inline lib/supplejack/request.rb]: Does RestClient require an input hash with HashRockets?

Comment (UberMouse) [inline lib/supplejack/request.rb]: No, I just copy pasted the other request methods

File: lib/supplejack/story.rb [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+    end
+
+    # Returns a comma separated list of tags for this Story
+    #
+    def tag_list
+      self.tags.join(', ') if self.tags
+    end
+
+    def private?
+      self.privacy == "private"
```
Comment (EdwinRozario) [inline lib/supplejack/story.rb]: Single Quote from string.

File: lib/supplejack/user_story_relation.rb [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+      @stories = if force
+                   fetch_stories.call
+                  elsif !@initial_fetch
+                    @initial_fetch = true
+                    fetch_stories.call
+                  else
+                    @stories
+                  end
+
+      @stories
```
Comment (EdwinRozario) [inline lib/supplejack/user_story_relation.rb]: Is this method returning @stories and also updating @stories?

Comment (UberMouse) [inline lib/supplejack/user_story_relation.rb]: Potentially doesn't need `@stories` at the end, but yes

**Comment by EdwinRozario** [comment]
👍

File: spec/supplejack/story_spec.rb [hunk 4]
```diff
@@ -0,0 +1,389 @@
...
+
+      it 'handles nil attributes' do
+        expect{Supplejack::Story.new(nil).attributes}.not_to raise_error
+      end
+
+      it 'initializes a user object' do
+        story = Supplejack::Story.new({user: {name: 'Juanito'}})
+        expect(story.user).to be_a Supplejack::User
+        expect(story.user.name).to eq 'Juanito'
+      end
```
Comment (ostigley) [inline spec/supplejack/story_spec.rb]: Juanito for the win

**Pull request merged**